docs/core/extensions/httpclient-http3.md

```diff
@@ -0,0 +1,116 @@
+>
+> Apps configured to take advantage of HTTP/3 should be designed to also support HTTP/1.1 and HTTP/2. If issues are identified in HTTP/3, we recommended disabling HTTP/3 until the issues are resolved in a future release of .NET.
+
+## HTTP/3 requirements
+
+### HttpClient Settings
+
+HTTP/3 support is in preview, and needs to be enabled via a configuration flag which can be set in the project with:
+
+```xml
+<ItemGroup>
+    <RuntimeHostConfigurationOption Include="System.Net.SocketsHttpHandler.Http3Support" Value="true">
+</ItemGroup>
+```
+
+Or using [`AppContext.SetSwitch`](/dotnet/api/system.appcontext.setswitch).
+
+The HTTP version can be configured by setting `HttpRequestMessage.Version` to 3.0. However, because not all routers, firewalls, and proxies properly support HTTP/3, we recommend configuring HTTP/3 together with HTTP/1.1 and HTTP/2. In HttpClient, this can be done by specifying:
+
+- `HttpRequestMessage.Version` to 1.1.
+- `HttpRequestMessage.VersionPolicy` to `HttpVersionPolicy.RequestVersionOrHigher`.
+
+The reason for requiring a configuration flag for HTTP/3 is to protect apps from future breakage when using version policy `RequestVersionOrHigher`. When calling a server that currently uses HTTP/1.1 and HTTP/2, if the server later upgrades to HTTP/3, the client would try to use HTTP/3 and potentially be incompatible as the standard is not final and therefore may change after .NET 6 is released.
+
+### Platform Dependencies
+
+HTTP/3 uses QUIC as its transport protocol. The .NET implementation of HTTP/3 uses [MsQuic](https://github.com/microsoft/msquic) to provide QUIC functionality. MsQuic is included in specific builds of windows and as a library for Linux. If the platform that HttpClient is running on doesn't have all the requirements for HTTP/3 then it's disabled.
+
+**Windows**
+
+- Windows 11 Build 22000 or later.
+- TLS 1.3 or later connection.
+
+The preceding Windows 11 Build versions may require the use of a [Windows Insider](https://insider.windows.com/) build.
+
+**Linux**
+
+On Linux, libmsquic is published via Microsoft official Linux package repository packages.microsoft.com. In order to consume it, it must be added manually. See [Linux Software Repository for Microsoft Products](https://windows-server/administration/linux-package-repository-for-microsoft-software). After configuring the package feed, it can be installed via the package manager of your distro, for example, for Ubuntu:
+
+```shell
+sudo apt install libmsquic
+```
+
+**macOS**
+
+HTTP/3 is not currently supported on macOS but may be available in a future release.
+
+### Get started
+
+Include the following in the project file to enable HTTP/3 with HttpClient:
+
+```xml
+<ItemGroup>
+    <RuntimeHostConfigurationOption Include="System.Net.SocketsHttpHandler.Http3Support" Value="true" />
+</ItemGroup>
+```
+
+HTTP3 needs to be specified as the version for the request:
+
+```c#
```

Comment:
Also, can the code for this be included in a snippet project and then referenced? This way it will be compiled and ensure it's clean code.

You would create the [following folder structure](https://docs.microsoft.com/en-us/contribute/dotnet/dotnet-contribute#example-folder-structure) inside of this folder:
```
./snippets/httpclient-http3/csharp
```

And put your code project there. Please see this documentation for how to define a snippet in your code and how to reference it. It's pretty simple. https://review.docs.microsoft.com/en-us/help/contribute/code-in-docs?branch=master#named-snippet I believe this is using top-level statements, so you wouldn't need a snippet id, just reference to the program.cs file.